tests/integration/plugins/test_python.py

```diff
@@ -408,3 +409,54 @@ def _get_package_install_commands(self) -> list[str]:
 
     primed_script = lf.project_info.prime_dir / "bin/mytest"
     assert not primed_script.exists()
+
+
+@pytest.mark.slow
+def test_python_plugin_get_files(new_dir, partitions):
+    parts_yaml = textwrap.dedent(
+        """\
+        parts:
+          foo:
+            plugin: python
+            source: .
+            python-packages: [flask==3.1.0]
+        """
+    )
+    parts = yaml.safe_load(parts_yaml)
+
+    lifecycle = LifecycleManager(
+        parts,
+        application_name="test_python",
+        cache_dir=new_dir,
+        partitions=partitions,
+    )
+    actions = lifecycle.plan(Step.BUILD)
+
+    with lifecycle.action_executor() as ctx:
+        ctx.execute(actions)
+
+    part_name = list(parts["parts"].keys())[0]
+    actual_file_list = lifecycle._executor._handler[part_name]._plugin.get_files()
+    part_install_dir = lifecycle._executor._part_list[0].part_install_dir
+
+    # Make sure all the expected packages were installed.
+    # We can't assert the exact set of keys because the pip version will change
+    # over time.  And we can't assert the number of keys because py3.10 installs
+    # setuptools as a separate package.
+    print(actual_file_list.keys())
+    for expected_pkg in {
```

Comment:
won't this fail as soon as a new version of any of these packages comes out?

Reply:
https://github.com/canonical/craft-parts/pull/943/commits/344dc5ac2cd17855c2d6a72fc9216cafc17f33fe